src/compas/files/base_reader.py

```diff
@@ -0,0 +1,188 @@
+        return self._is_url
+
+    def download(self, url):
+        """Downloads file and returns path to tempfle
+
+        Called by property self.location
+
+        Parameters
+        ----------
+        url : string
+            URL to file
+
+        Returns
+        -------
+        location : Pathlib object
+            Path to tempfile
+        """
+
+        location, _ = urlretrieve(url)
+
+        return Path(location)
+
+    def open_ascii(self):
+        """Open ascii or binary file and return file object
+
+        Returns
+        -------
+        file object
+        """
+        try:
+            file_object = self.location.open(mode='r')
+        # unnecessary now?
+        except UnicodeDecodeError:
+            file_object = self.location.open(mode='r', errors='replace', newline='\r')
+        return file_object
+
+    def open_binary(self):
+        """Open binary file and return file object
+
+        Returns
+        -------
+        file object
+        """
+
+        return self.location.open(mode='rb')
+
+    def iter_lines(self):
+        """Yields lines from local ascii files
+
+        Yields
+        -------
+        string
+            Next line or chunk of file
+        """
+        # TODO: Handle continuing lines (as in OFF files)
+
+        with self.open_ascii() as fo:
+            for line in fo:
+                yield line.rstrip()
+
+    def iter_chunks(self, chunk_size=1024):
+        """Yields lines from local binary files
+
+        Parameters
+        ----------
+        chunk_size : int
+            Chunks to read with each call
+
+        Yields
+        ------
+        bytes
+            Next chunk of file
+        """
+        raise NotImplementedError
+
+    def read(self):
+        raise NotImplementedError
+
+    def check_file_signature(self):
+        """Checks wether file signature (also known as magic number) is present
+        in input file.
+
+        File signatures are strings, numbers or bytes defined in a file
+        format's specification. While not technically required to parse the
+        file, a missing file signatures might be a sign of a malformed file.
+
+        More information about file signatures can be found on Wikipedia[1]_
+        as well as examples of file signatures[2]_
+
+        Raises
+        ------
+        Exception
+            If file signature is specified for file format but not present
+            in file.
+
+        .. [1] https://en.wikipedia.org/wiki/List_of_file_signatures
+        .. [2] https://en.wikipedia.org/wiki/File_format#Magic_number
+        """
+        # TODO: Integrate with is_binary checks in file readers
+        # TODO. Look into binaryornot library to help with above
+
+        try:
+            file_signature = self.FILE_SIGNATURE['content']
+            signature_offset = self.FILE_SIGNATURE['offset']
+
+        # if file type has no associated file signature
+        except AttributeError:
+            return
```

Comment:
the `BaseReader` should define defaults for both parameters.
```python
class BaseReader(object):
    FILE_SIGNATURE = {
         'content': None,
         'offset': None,
    }
```
after which the above becaomes a simple if statement.